src/core/thread/dua_manager.cpp

```diff
@@ -184,6 +242,498 @@ otError DuaManager::Store(void)
+}
+
+void DuaManager::ScheduleTimer(void)
+{
+    if (mDelay.mValue == 0)
+    {
+        mTimer.Stop();
+    }
+    else if (!mTimer.IsRunning())
+    {
+        mTimer.Start(kStateUpdatePeriod);
+    }
+}
+
+void DuaManager::PerformNextRegistration(void)
+{
+    otError          error   = OT_ERROR_NONE;
+    Mle::MleRouter & mle     = Get<Mle::MleRouter>();
+    Coap::Message *  message = nullptr;
+    Ip6::MessageInfo messageInfo;
+    Ip6::Address     dua;
+
+    VerifyOrExit(mle.IsAttached(), error = OT_ERROR_INVALID_STATE);
+    VerifyOrExit(Get<BackboneRouter::Leader>().HasPrimary(), error = OT_ERROR_INVALID_STATE);
+
+    // Only allow one outgoing DUA.req
+    VerifyOrExit(!mIsDuaPending, error = OT_ERROR_BUSY);
+
+    // Only send DUA.req when necessary
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    VerifyOrExit(mle.IsRouterOrLeader() || !mle.IsExpectedToBecomeRouter(), error = OT_ERROR_INVALID_STATE);
+    VerifyOrExit((mDuaState == kToRegister && mDelay.mFields.mRegistrationDelay == 0) ||
+                     mChildDuaMask != mChildDuaRegisteredMask,
+                 error = OT_ERROR_NOT_FOUND);
+#else
+    VerifyOrExit(mDuaState == kToRegister && mDelay.mFields.mRegistrationDelay == 0, error = OT_ERROR_NOT_FOUND);
+#endif // OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+
+    VerifyOrExit(mle.IsFullThreadDevice() || mle.GetParent().IsThreadVersion1p1(), error = OT_ERROR_INVALID_STATE);
+#endif // OPENTHREAD_CONFIG_DUA_ENABLE
+
+    // Prepare DUA.req
+    VerifyOrExit((message = Get<Coap::Coap>().NewPriorityMessage()) != nullptr, error = OT_ERROR_NO_BUFS);
+
+    SuccessOrExit(error =
+                      message->Init(OT_COAP_TYPE_CONFIRMABLE, OT_COAP_CODE_POST, OT_URI_PATH_DUA_REGISTRATION_REQUEST));
+    SuccessOrExit(error = message->SetPayloadMarker());
+
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    if (mDuaState == kToRegister && mDelay.mFields.mRegistrationDelay == 0)
+    {
+        dua = GetDomainUnicastAddress();
+        SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kTarget, &dua, sizeof(dua)));
+        SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kMeshLocalEid, &mle.GetMeshLocal64().GetIid(),
+                                             sizeof(Ip6::InterfaceIdentifier)));
+        mDuaState             = kRegistering;
+        mLastRegistrationTime = TimerMilli::GetNow();
+    }
+    else
+#endif // OPENTHREAD_CONFIG_DUA_ENABLE
+    {
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+        if (!mRegisterCurrentChildIndex)
+        {
+            for (Child &child : Get<ChildTable>().Iterate(Child::kInStateValid))
+            {
+                uint16_t childIndex = Get<ChildTable>().GetChildIndex(child);
+
+                if (mChildDuaMask.Get(childIndex) && !mChildDuaRegisteredMask.Get(childIndex))
+                {
+                    mChildIndexDuaRegistering = childIndex;
+                    break;
+                }
+            }
+        }
+        {
```

Comment:
This can easily misleading a reader that it looks very like the `else-statement`.

Reply:
adjusted a bit to remove the bracket here.